### PR TITLE
Move/service icons to js components package

### PR DESCRIPTION
**Pull request by mkm119**
#### Changes proposed in this Pull Request:
Moved SocialServiceIcon component from Jetpack Icons.js file to js-package/components. Updated it's ref in the Jetpack plugin directory

#### Testing instructions:

* Make sure `pnpm build-extensions` is running fine in Jetpack plugin directory
* Add a new social connection in Marketing>Connections dashboard
* Create a new post and open the Jetpack sidebar from the top left to share this post. Ensure that the connected service icon is getting rendered properly

Screenshot 2022-04-05 at 9 18 45 PM

### CODE REVIEW

File: projects/js-packages/components/components/social-service-icon/style.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// TODO: @mkm119 replace these with gutenber-base-styles once that file is moved to js-packages/base-styles
```
Comment (pablinos) [inline projects/js-packages/components/components/social-service-icon/style.scss:1]: I wonder if we should wait for that and update it in this PR. If we can get that file in `base-styles` over the next couple of days, it won't hold me up. I'm not sure if there's anything else that is waiting on this icon component.

**Review by pablinos — COMMENTED** [review]
This is looking good to me. We can decide if we want to wait for the base styles change or not.

File: projects/js-packages/components/components/social-service-icon/style.scss (hunk 1 above)
Comment (mkm119) [reply projects/js-packages/components/components/social-service-icon/style.scss:1]: I can update this as part of the next PR if this gets merged before that one @pablinos

File: projects/js-packages/base-styles/gutenberg-base-styles.scss [hunk 2]
```diff
@@ -1,6 +1,6 @@
-@import '@wordpress/base-styles/z-index';
-@import '@wordpress/base-styles/colors';
-@import '@wordpress/base-styles/variables';
-@import '@wordpress/base-styles/breakpoints';
-@import '@wordpress/base-styles/mixins';
-@import '@wordpress/base-styles/animations';
+@import '~@wordpress/base-styles/z-index';
```
Comment (pablinos) [inline projects/js-packages/base-styles/gutenberg-base-styles.scss:1]: I've had to add the tildes here as Boost is using Rollup, and it doesn't resolve the scss files from `node_modules` without it.

p1651747887163089-slack-CDLH4C1UZ

**Review by retrofox — COMMENTED** [review]
Hi, there.
It could be nice if we could add a story for this implementation. something like [Product Icons Story](https://automattic.github.io/jetpack-storybook/?path=/story/js-packages-components-product-icons--default)
Also, the approach here seems to be slightly different from [Product Icons](https://github.com/Automattic/jetpack/blob/master/projects/js-packages/components/components/product-icons/index.jsx). 

Finally and thinking in the medium long term, I think we should try to follow the current core implementation, which uses the Icon component:

```jsx
import { Icon, chartBar } from '@wordpress/icons';

function MyComponent() {
  return <Icon icon={ chartBar } />
}
```

**Comment by pablinos** [comment]
Cheers @retrofox! Yeah this is just copying stuff over to the components package, so it's probably using some old patterns. I'll see about updating it.

**Comment by retrofox** [comment]
> Cheers @retrofox! Yeah this is just copying stuff over to the components package, so it's probably using some old patterns. I'll see about updating it.

👋 I don't see a big deal, actually. It seems to be a matter of exporting [IconWrapper](https://github.com/Automattic/jetpack/blob/master/projects/js-packages/components/components/product-icons/index.jsx/#L24) to be used by the new icons. For instance, the [AntiSpamIcon](https://github.com/Automattic/jetpack/blob/6317f23198187e0595b678d9136e14c2fd398164/projects/js-packages/components/components/product-icons/index.jsx#L48) component.

File: projects/js-packages/components/components/icons/style.module.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
+@import '@automattic/jetpack-base-styles/gutenberg-base-styles';
```
Comment (pablinos) [inline projects/js-packages/components/components/icons/style.module.scss:1]: I'm not sure whether this is still relevant with the way we're importing styles. Everywhere we're using this, `gutenberg-base-styles` will be included elsewhere.

We could maybe add this to the main package stylesheet instead?

Comment (retrofox) [reply projects/js-packages/components/components/icons/style.module.scss:1]: I'm not sure about this. @renatoagds maybe can help us here

**Comment by pablinos** [comment]
> could you rename the [story name](https://github.com/Automattic/jetpack/blob/7385a51bdfa5f348242854e616f402f3f88b0328/projects/js-packages/components/components/product-icons/stories/index.jsx#L13)? Provably we'd like to update [the doc](https://github.com/Automattic/jetpack/blob/7385a51bdfa5f348242854e616f402f3f88b0328/projects/js-packages/components/components/product-icons/README.md) too.

Good point. I thought I'd done that. Addressed in [6ff614f](https://github.com/Automattic/jetpack/pull/23795/commits/6ff614f040b6d8bf707dcc42d6b297570836fda7)

**Review by spsiddarthan — DISMISSED** [review]

**Review by samiff — DISMISSED** [review]
Tests well for me

File: projects/js-packages/components/components/icons/style.module.scss (hunk 3 above)
Comment (renatoagds) [reply projects/js-packages/components/components/icons/style.module.scss:1]: @pablinos Yes, it is better to remove it from here. This kind of style import should happen only at consumers (`plugin` or `package`).

File: projects/js-packages/components/components/icons/index.jsx [hunk 4]
```diff
@@ -26,20 +27,22 @@ function IconWrapper( {
 	size = 24,
 	viewBox = '0 0 24 24',
 	opacity = 1,
-	color = '#2C3338',
+	color,
```
Comment (renatoagds) [inline projects/js-packages/components/components/icons/index.jsx:30]: @pablinos Removing the default probably will break some current usages, not? Any reason to do it?

File: projects/js-packages/components/components/icons/index.jsx [hunk 5]
```diff
@@ -159,3 +261,11 @@ export function getIconBySlug( slug ) {
 
 	return iconsMap[ slug ];
 }
+
+export const SocialServiceIcon = ( { serviceName, className } ) => {
+	const Icon = getIconBySlug( serviceName );
+	if ( Icon ) {
+		return <Icon className={ className } />;
+	}
+	return null;
```
Comment (renatoagds) [inline projects/js-packages/components/components/icons/index.jsx:270]:
```es6
return Icon ? <Icon className={ className } /> : null
```

File: projects/js-packages/components/components/icons/style.module.scss [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	}
+	&.linkedin {
+		fill: var( --color-linkedin );
+	}
+	&.tumblr {
+		fill: var( --color-tumblr );
+	}
+	&.google {
+		fill: var( --color-gplus );
+	}
```
Comment (renatoagds) [inline projects/js-packages/components/components/icons/style.module.scss:32]: Now, we don't have these `CSS Vars.` Suggest to hardcode it for now, and we can move to CSS Vars in some moment.

File: projects/js-packages/components/components/icons/index.jsx (hunk 4 above)
Comment (pablinos) [reply projects/js-packages/components/components/icons/index.jsx:30]: With the colour prop it sets the `fill` attribute and makes overriding it difficult, so instead [I set the default colour in the CSS](https://github.com/Automattic/jetpack/pull/23795/files#diff-60b2cde8f5df00ff59380f078572b8ec302053a978eaa840c152ee1567d36a01R4)

File: projects/js-packages/components/components/icons/index.jsx (hunk 5 above)
Comment (pablinos) [reply projects/js-packages/components/components/icons/index.jsx:270]: I think I just copied this over from the existing Product Icons code. I'll update this though.

File: projects/js-packages/components/components/icons/style.module.scss (hunk 6 above)
Comment (pablinos) [reply projects/js-packages/components/components/icons/style.module.scss:32]: There's a PostCSS script that substitutes these in Jetpack. It's a bit weird, and I didn't want to break something else.

**Comment by pablinos** [comment]
@samiff @renatoagds - I've made some changes based on feedback and having to rebase this. It's causing updates in most of the projects in the repo, so I'd love to get it merged before another change the package gets in there!

I've had to do a fair amount of nursing to get D78213-code happy too, but I think that's good to merge if this is.

@spsiddarthan Would you mind keeping an eye on this tomorrow, and merging everything if it gets approved?

**Review by samiff — APPROVED** [review]

**Comment by pablinos** [comment]
r245583-wpcom

**Pull request merged**